### PR TITLE
Reorganize module hierarchy

### PR DESCRIPTION
**Pull request by kozross**
This is the first attempt to try and make some sense of the module hierarchy. The overall goals were:

1. Users should need to only import `Plutarch.Prelude` and get everything they need
2. Plutarch itself shouldn't import `Plutarch.Prelude`, only things in the `Plutarch.Internal.*` namespace
3. `plutarch-ledger-api` and `plutarch-testlib` should rely minimally on imports from `Plutarch.Internal.*`

Additionally, there were a couple of changes:

1. All builtins should have a `pbuiltinMyBuiltinName` in `Plutarch.Internal.Builtin`, allowing them to be called without having to wonder what their signatures are.
2. `Plutarch.*` has been significantly emptied.
3. `Plutarch.Internal.Builtin` contains (most of) the types that we _have_ to have because they're mandated by Plutus.
4. Some modules got merged.

This is not the final outcome, I just wanted to get _something_ reviewed and merged, or the diffs would get too huge.

### CODE REVIEW

File: src/Plutarch/Internal/Builtin.hs [hunk 1]
```diff
@@ -0,0 +1,810 @@
+{-# LANGUAGE FlexibleInstances #-}
```
Comment (SeungheonOh) [inline src/Plutarch/Internal/Builtin.hs:1]: This is quite massive. More so than I thought, would it be better to make `Plutarch.Builtins.XYZ` modules instead?

Comment (kozross) [reply src/Plutarch/Internal/Builtin.hs:1]: Yeah, I agree.

**Comment by SeungheonOh** [comment]
Will also resolve #735

**Comment by SeungheonOh** [comment]
Notes: this is closed because of nasty conflict

**Pull request closed without merging**